nixos/modules/services/web-apps/kanboard.nix

```
@@ -0,0 +1,137 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.kanboard;
+
+  toStringAttrs = attrs: lib.attrsets.mapAttrs (k: v: toString v) attrs;
+in
+{
+  meta.maintainers = with lib.maintainers; [ yzx9 ];
+
+  options.services.kanboard = with lib; {
```

```suggestion
  options.services.kanboard = {
```
`with lib;` [isn't very good practice](https://nix.dev/guides/best-practices#with-scopes). You should instead bring these functions into scope by using `inherit` or use the fully qualified names (i.e., `lib.mkOption`)

---

I’ve simplified most of the `with lib`, except for with lib.types, because I believe using with `lib.types` within a limited scope keeps things simple and unambiguous. However, I’m happy to adjust if you think we should strictly follow best practices.

Use:

```nix
with lib.types;
attrsOf (oneOf [
  str
  int
  bool
]);
```

Not use:

```nix
lib.types.attrsOf (
  lib.types.oneOf [
    lib.types.str
    lib.types.int
    lib.types.bool
  ]
);
```

---

This is totally fine. `lib.types` is a much narrower scope than all of `lib`, and there aren't any other namespaces being referred to here -- which is like the perfect case to use `with ...;`